src/Icicle/Source/Type/Constraints.hs

```diff
@@ -94,15 +94,18 @@ dischargeC c
      | btemp == ctemp
      -> dischargeC $ CEquals atemp btemp
 
-    CTemporalityJoin (TypeVar _) _          (TypeVar _)
+    CTemporalityJoin (TypeVar _) _ (TypeVar _)
      -> return $ DischargeLeftover c
     CTemporalityJoin (TypeVar _) (TypeVar _) _
```

Comment:
actually, can probably remove these cases now - they'll be covered by the two new ones